tests/samplers_tests/tpe_tests/test_sampler.py

```diff
@@ -26,20 +27,336 @@ def test_hyperopt_parameters(use_hyperband: bool) -> None:
     study.optimize(lambda t: t.suggest_uniform("x", 10, 20), n_trials=50)
 
 
-def test_sample_relative() -> None:
+def test_multivariate_experimental_warning() -> None:
+    with pytest.warns(optuna.exceptions.ExperimentalWarning):
+        optuna.samplers.TPESampler(multivariate=True)
+
+
+def test_infer_relative_search_space() -> None:
+
     sampler = TPESampler()
+    search_space = {
+        "a": distributions.UniformDistribution(1.0, 100.0),
+        "b": distributions.LogUniformDistribution(1.0, 100.0),
+        "c": distributions.DiscreteUniformDistribution(1.0, 100.0, 3.0),
+        "d": distributions.IntUniformDistribution(1, 100),
```

Comment:
How about adding an int distribution with `step=3` (or something) to the search space? I think the tests don't contain `trial.suggest_int` with the step parameter.

Reply:
I will add a test. Thank you for the comment.